File: apps/mobile/src/services/livekit.ts
/**
 * LiveKit Service — Manages LiveKit tokens and connection details.
 */

import { apiClient } from "./api-client";
import { ENV } from "../config/env";

export async function getLiveKitToken(meetingId: string, identity: string, name: string, email?: string | null) {
  const data = await apiClient.post<{ token: string }>(
    `/api/livekit/token`,
    {
      roomName: meetingId,
      identity,
      name,
      email,
    }
  );
  return data.token;
}

File: apps/backend/app/schemas/livekit.py
"""
Pydantic schemas for the LiveKit token endpoint.
"""

from pydantic import BaseModel, Field


class TokenRequest(BaseModel):
    """Request body for POST /api/livekit/token."""

    roomName: str = Field(
        ...,
        min_length=1,
        max_length=128,
        description="LiveKit room name (typically the Firestore meeting ID).",
    )
    identity: str = Field(
        ...,
        min_length=1,
        max_length=128,
        description="Participant identity (typically the Firebase user UID).",
    )
    name: str = Field(
        ...,
        min_length=1,
        max_length=256,
        description="Participant display name.",
    )
    email: str | None = Field(
        default=None,
        description="Participant email.",
    )


class TokenResponse(BaseModel):
    """Response body for POST /api/livekit/token."""

    token: str = Field(..., description="Signed LiveKit access token (JWT).")
